libs/isograph-react/src/loadable-hooks/usePagination.ts

```diff
@@ -0,0 +1,315 @@
+            This is indicative of a bug in useSkipLimitPagination.',
+          );
+        }
+        return clonedRefCountedPointer;
+      });
+      clonedPointers.push(newPointer);
+
+      const totalItemCleanupPair: ItemCleanupPair<
+        ReadonlyArray<
+          ItemCleanupPair<
+            ReferenceCountedPointer<
+              FragmentReference<TReadFromStore, Connection<TItem>>
+            >
+          >
+        >
+      > = [
+        clonedPointers,
+        () => {
+          clonedPointers.forEach(([, dispose]) => {
+            dispose();
+          });
+        },
+      ];
+
+      setState(totalItemCleanupPair);
+    };
+
+  const [, rerender] = useState({});
+
+  const loadedReferences = state === UNASSIGNED_STATE ? [] : state;
+
+  const mostRecentItem: LoadedFragmentReference<
+    TReadFromStore,
+    Connection<TItem>
+  > | null = loadedReferences[loadedReferences.length - 1];
+  const mostRecentFragmentReference =
+    mostRecentItem?.[0].getItemIfNotDisposed();
+
+  if (mostRecentItem && mostRecentFragmentReference === null) {
+    throw new Error(
+      'FragmentReference is unexpectedly disposed. \
+      This is indicative of a bug in Isograph.',
+    );
+  }
+
+  const networkRequestStatus =
+    mostRecentFragmentReference &&
+    getPromiseState(mostRecentFragmentReference.networkRequest);
+
+  const slicedFragmentReferences =
+    networkRequestStatus?.kind === 'Ok'
+      ? loadedReferences
+      : loadedReferences.slice(0, loadedReferences.length - 1);
+
+  const completedFragmentReferences = slicedFragmentReferences.map(
+    ([pointer]) => {
+      const fragmentReference = pointer.getItemIfNotDisposed();
+      if (fragmentReference == null) {
+        throw new Error(
+          'FragmentReference is unexpectedly disposed. \
+            This is indicative of a bug in Isograph.',
+        );
+      }
+      return fragmentReference;
+    },
+  );
+
+  const readOutDataAndRecords = completedFragmentReferences.map(
+    (fragmentReference) =>
+      readButDoNotEvaluate(
+        environment,
+        fragmentReference,
+        networkRequestOptions,
+      ),
+  );
+
+  useSubscribeToMultiple<TReadFromStore>(
+    subscribeCompletedFragmentReferences(completedFragmentReferences),
+  );
+
+  if (!networkRequestStatus) {
+    return {
+      kind: 'Complete',
+      fetchMore: getFetchMore(null),
+      results: [],
+      hasNextPage: true,
+    };
+  }
+
+  switch (networkRequestStatus.kind) {
+    case 'Pending': {
+      const unsubscribe = subscribeToAnyChange(environment, () => {
+        unsubscribe();
+        rerender({});
+      });
+
+      const results = readCompletedFragmentReferences(
+        completedFragmentReferences,
+      );
+      return {
+        results: results.edges,
+        kind: 'Pending',
+        //@ts-expect-error map Connection<TItem> to ReadonlyArray<TItem>
+        pendingFragment: mostRecentFragmentReference,
```

Comment:
How do you feel about this, is it possible to map the fragment to return just the edges, or should it return whole `Connection`?

Reply:
I would probably map it, so that it's consistent with the shape of `results`. I'm not 1000% sure.